internal/cortex/querier/queryrange/query_range.go

```diff
@@ -211,22 +209,38 @@ func NewEmptyPrometheusInstantQueryResponse() *PrometheusInstantQueryResponse {
 	}
 }
 
-type TimeDurationTransformer struct{}
+func traverseAnalysis(a *Analysis, results *[]*Analysis) {
+	if a == nil {
+		return
+	}
 
-func (t TimeDurationTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
-	if typ == reflect.TypeOf(Duration(0)) {
-		return func(dst, src reflect.Value) error {
-			if dst.CanSet() {
-				d := dst.Interface().(Duration)
-				s := src.Interface().(Duration)
+	*results = append(*results, a)
 
-				merged := d + s
-				dst.Set(reflect.ValueOf(merged))
-			}
-			return nil
+	for _, ch := range a.Children {
+		traverseAnalysis(ch, results)
+	}
+}
+
+func AnalyzesMerge(analysis ...*Analysis) *Analysis {
```

Comment:
Might be a slightly more appropriate name

```suggestion
func MergeAnalyses(analysis ...*Analysis) *Analysis {
```